Articles/Blog/2023-05-sci-workflows.md

```diff
@@ -0,0 +1,93 @@
+
+Taken together, these properties meant that it took us months to implement a scalable application linking MUQ and ExaHyPE. We had to dig deep into the parallelization of ExaHyPE for it to accept MPI sub-communicators from MUQ, modify its main routines to call it as a library, and rewrite parts of its custom build system. Along the way, we encountered a myriad of unexpected issues.
+
+### Microservices in scientific software
+
+Linking MUQ and ExaHyPE was a painful experience, contrary to the theoretically simple link of the algorithms.
+
+We, therefore, decided to try a completely different approach, taking inspiration from microservice architectures. Enter UM-Bridge, the UQ and Modeling Bridge: Instead of building one monolithic application encompassing UQ, numerical model, and HPC capabilities, we now separate each aspect into its own component. The UQ application requests evaluations through the UM-Bridge network protocol, while the model application is now a server, launching simulation runs upon request. The model may run on a cluster, where a load balancer takes care of many parallelization aspects.
+
+<br> 
+
+<img src='../../images/sci-workflows-4.png' class='page lightbox' />[The software now mirrors the abstract theoretical interface between UQ algorithm and model.]
+
+<br>
+
+Right away, UM-Bridge offers a number of benefits:
+
+* We can easily link tools that are written in different programming languages or are otherwise incompatible; all we need now is a small extension to support UM-Bridge on either side, which is easy to achieve using one of the language-specific or even framework-specific integrations we provide.
+
+* UM-Bridge models can readily be containerized since the natural path to accessing a containerized application is via a network. Containerized models can be shared among collaborators, improving the separation of concerns between UQ and model experts. Containers also provide a high degree of reproducibility, which we used to build the first library of ready-to-run UQ benchmark problems.
+
+* Simple thread-parallel UQ codes are now enough to offload parallel model runs to a cluster since distributing work across many model instances on the cluster is now up to UM-Bridge (specifically the Kubernetes setup we provide).
+
+### Rapid UQ application development: tsunamis revisited
+
+We set out to revisit the tsunami problem using UM-Bridge. As before, we went with the ExaHyPE tsunami model. This time, we chose multilevel delayed acceptance (MLDA) as our UQ method. To spice things up, we wanted to build a Gaussian Process (GP) surrogate as the coarsest level in MLDA, replacing our less accurate hand-crafted coarse approximation. We used Google Cloud Platform (GCP) to provide the computational horsepower.
+
+We had a clear separation of roles: Anne is the ExaHyPE expert, Mikkel is the developer of MLDA, and Linus has the most experience in cloud computing. UM-Bridge facilitated development at all stages:
```

Comment:
I wonder if it would be appropriate to include Mikkel as a co-author?  If not, it looks a bit odd to refer to someone without having any indication of who they really are.  An alternative might be something like "Anne is the ExaHype expert, Linus has the most..., and our collaborator Mikkel Lastname is the developer of MLDA."

Reply:
Thanks a lot for spotting this! Mikkel hasn't been involved with the design of UM-Bridge, so I went with the second option you suggested.